runtime/gc_structs/ConstantPoolObjectSlotIterator.hpp

```diff
@@ -48,18 +49,44 @@ class GC_ConstantPoolObjectSlotIterator
 	U_32 *_cpDescriptionSlots;
 	U_32 _cpDescription;
 	UDATA _cpDescriptionIndex;
+	bool _condyOnly;
+	bool _condyEnabled;
 	
+private:
+	/**
+	 * Scan the current constant dynamic slot
+	 * @return slot value if it is not NULL
+	 * @return exception otherwise
+	 */
+	MMINLINE j9object_t* scanCondySlot(j9object_t *slotPtr)
+	{
+		j9object_t * result;
+		if (NULL != ((J9RAMConstantDynamicRef *) slotPtr)->value) {
+			result = &(((J9RAMConstantDynamicRef *) slotPtr)->value);
+		} else {
+			result = &(((J9RAMConstantDynamicRef *) slotPtr)->exception);
+		}
```

Comment:
Did GAC's comment about both slots needing to be scanned get addressed?  This looks like it only scans 1 or the other

Reply:
@DanHeidinga not yet 

@Ailloviee  you'll need to add  additional state information to return two slots from one cpentry

Reply:
The code is technically correct at the moment, but it's a timebomb waiting to go off.